cron/centAcl.php

```diff
@@ -199,19 +214,29 @@ function programExit($msg)
          * Add Hostgroups
          */
         if ($row['all_hostgroups']) {
-            $query = "SELECT hg_id FROM hostgroup WHERE hg_id NOT IN (SELECT DISTINCT hg_hg_id " .
-                "FROM acl_resources_hg_relations WHERE acl_res_id = '" . $row['acl_res_id'] . "')";
-            $res1 = $pearDB->query($query);
+            $res1 = $pearDB->prepare(
+                "SELECT hg_id FROM hostgroup
+                WHERE hg_id NOT IN (
+                    SELECT DISTINCT hg_hg_id FROM acl_resources_hg_relations
+                    WHERE acl_res_id = :aclResId)"
+            );
+            $res1->bindValue(':aclResId', $row['acl_res_id'], \PDO::PARAM_INT);
+            $res1->execute();
+
 
             if ($res1->rowCount()) {
                 // set acl_resources.changed flag to 1
                 $aclResourcesUpdated = true;
             }
 
             while ($rowData = $res1->fetch()) {
-                $insert_query = "INSERT INTO acl_resources_hg_relations (hg_hg_id, acl_res_id) VALUES ('" .
```

Comment:
Don't forget to create a db transaction when there is more than one update request. (=> while)